src/react-components/profile-entry-panel.js

```diff
@@ -62,28 +82,27 @@ class ProfileEntryPanel extends Component {
     e.stopPropagation();
   };
 
-  setAvatarStateFromIframeMessage = e => {
-    if (e.source !== this.avatarSelector.contentWindow) {
-      return;
-    }
-    this.setState({ avatarId: e.data.avatarId });
-
-    // Send it back down to cause avatar picker UI to update.
-    this.sendAvatarStateToIframe();
-  };
-
-  sendAvatarStateToIframe = () => {
-    this.avatarSelector.contentWindow.postMessage({ avatarId: this.state.avatarId }, location.origin);
+  setAvatarFromMediaResult = ({ detail: entry }) => {
+    this.setState({
+      avatarId: entry.id,
+      avatarType: getAvatarType(entry.id),
+      avatarGltfUrl: entry.gltfs.avatar
+    });
+    // Replace history state with the current avatar id since this component gets destroyed when we open the
```

Comment:
Confused why you can't just load the profile from localstorage like normal?

Reply:
This is referring to the ephemeral state, before you actually mutate localstorage.